tests/integration_tests/test_pytorch_distributed.py

```diff
@@ -351,3 +351,23 @@ def test_updates_properties(storage_mode: str) -> None:
             [getattr(trial, p) for p in property_names]
 
         dist.barrier()  # type: ignore[no-untyped-call]
+
+@pytest.mark.filterwarnings("ignore::optuna.exceptions.ExperimentalWarning")
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_pass_frozen_trial_to_torch_distributed(storage_mode: str) -> None:
+    # Regression test of #4697
+    def objective(trial):
+        trial = optuna.integration.TorchDistributedTrial(
+            trial if dist.get_rank() == 0 else None
+        )
```

Comment:
`black` suggests the follow change.
```suggestion
        trial = optuna.integration.TorchDistributedTrial(trial if dist.get_rank() == 0 else None)
```